napari/components/viewer_model.py

```diff
@@ -68,41 +82,47 @@ class ViewerModel(KeymapProvider, MousemapProvider):
         Contains axes, indices, dimensions and sliders.
     """
 
-    def __init__(self, title='napari', ndisplay=2, order=(), axis_labels=()):
-        super().__init__()
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            status=Event,
-            help=Event,
-            title=Event,
-            reset_view=Event,
-            active_layer=Event,
-            theme=Event,
-            layers_change=Event,
-        )
+    # After https://github.com/samuelcolvin/pydantic/pull/2196 is released
+    # make all of these fields with allow_mutation=False
+    axes: Axes = Axes()
+    camera: Camera = Camera()
+    cursor: Cursor = Cursor()
+    dims: Dims = Dims()
+    grid: GridCanvas = GridCanvas()
+    layers: LayerList = Field(
+        default_factory=LayerList
+    )  # Need to create custom JSON encoder for layer!
+    scale_bar: ScaleBar = ScaleBar()
+
+    active_layer: Optional[
+        Layer
+    ] = None  # Would be nice to remove this/ make it layer name instead of layer?
+    help: str = ''
+    status: str = 'Ready'
+    theme: str = DEFAULT_THEME
+    title: str = 'napari'
+
+    # 2-tuple indicating height and width
+    _canvas_size: Tuple[int, int] = (600, 800)
+
+    class Config(EventedModel.Config):
```

Comment:
I wasn't able to use the `self.__config__.extra = 'allow'` syntax in the init, it didn't seem to work for me. Not sure if I was getting something trivial wrong, but I tried with what you posted in comment above

Reply:
huh... what error did you get?  You can also try `self.__config__.extra = Extra.allow` with `from pydantic import Extra`

Reply:
Yup, that's it. Using just the string the extra attributes just don't get added, no special error message until you try and use one. Using the Enum works. I can use that syntax now